src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -697,6 +766,9 @@ void ImageReplayer<I>::on_stop_local_image_close_finish(int r)
   m_replay_status_formatter = nullptr;
 
   m_remote_journaler->stop_replay();
+  if (m_stopping_for_resync) {
+    m_remote_journaler->unregister_client();
```

Comment:
Wouldn't the ImageDeleter delete the image and thus the associated request_resync flag?